Include/cpython/pyatomic.h

```diff
@@ -0,0 +1,397 @@
+#ifndef Py_ATOMIC_H
+#define Py_ATOMIC_H
+
+// This header provides cross-platform low-level atomic operations
+// similar to C11 atomics.
+//
+// Operations are sequentially consistent unless they have a suffix indicating
+// otherwise. If in doubt, prefer the sequentially consistent operations.
+//
+// The "_relaxed" suffix for load and store operations indicates the "relaxed"
+// memory order. They don't provide synchronization, but (roughly speaking)
+// guarantee somewhat sane behavior for races instead of undefined behavior.
+// In practice, they correspond to "normal" hardware load and store instructions,
+// so they are almost as inexpensive as plain loads and stores in C.
+//
+// Note that atomic read-modify-write operations like _Py_atomic_add_* return
+// the previous value of the atomic variable, not the new value.
+//
+// See https://en.cppreference.com/w/c/atomic for more information on C11 atomics.
+// See https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2020/p2055r0.pdf
+// "A Relaxed Guide to memory_order_relaxed" for discussion of and common usage
+// or relaxed atomics.
+
+// Atomically adds `value` to `address` and returns the previous value
+static inline int
+_Py_atomic_add_int(int *address, int value);
```

Comment:
The glib library uses *atomic* for the first argument. Maybe it's a better name than *address* or *pointer*, what do you think?

Reply:
* https://en.cppreference.com/w/c/atomic uses ``obj`` name.
* https://www.ibm.com/docs/en/zos/2.1.0?topic=c11-atomic-load also uses *obj* name (C11 functions)
* pycore_atomic.h uses ``ATOMIC_VAL`` name
* pycore_atomic_func.h uses ``var`` name

I dislike the ``address`` name. What you pass is not a ``void*`` pointer or an ``uintptr_t`` address, but a reference to an atomic variable. I suggest to use *atomic* or *obj* name.

For add operation, the C11 API uses *arg* for the second parameter name. I'm fine with value.

Reply:
For context, the GCC documentation for built-in atomic functions uses `ptr`. This is the API that `pyatomic.h` most closely matches.

https://gcc.gnu.org/onlinedocs/gcc/_005f_005fatomic-Builtins.html